examples/ex3p.cpp

```diff
@@ -62,6 +62,8 @@ int main(int argc, char *argv[])
    MPI_Init(&argc, &argv);
    MPI_Comm_size(MPI_COMM_WORLD, &num_procs);
    MPI_Comm_rank(MPI_COMM_WORLD, &myid);
+   // HYPRE_Init();
```

Comment:
Is the plan to introduce these lines of code when hypre supports AMS on device?

Reply:
This is a reminder to come up with a unified way of initializing external libraries that need initialization and/or finalization.

Reply:
I think we should remove these from here, because they are not present in the other examples.

Alternatively, I can propose a PR to fix that, but I feel this will be a big change to introduce in mfem-4.3.

What do you think?

Reply:
The main motivation in `ex3p` was that without `HYPRE_SetSpGemmUseCusparse(0);` (which requires `HYPRE_Init();`) we get non-deterministic behavior with AMS. This may also affect BoomerAMG in other examples but I'm not sure.

Reply:
OK, we can then leave the comment for `HYPRE_SetSpGemmUseCusparse(0);` only and explain why it is there.

Reply:
I can make the above change, unless you feel we should introduce `MFEM_Session` in this PR

Reply:
Resolved by c754730